spyder/plugins/findinfiles/widgets/tests/test_widgets.py

```diff
@@ -310,6 +312,39 @@ def test_exclude_regexp_error(findinfiles, qtbot):
     assert findinfiles.REGEX_ERROR in tooltip
 
 
+@flaky(max_runs=5)
+def test_no_empty_file_items(findinfiles, qtbot):
+    """
+    Test that a search that hits the max number of results doesn't generate
+    empty file items.
+
+    This is a regression test for issue spyder-ide/spyder#16256
+    """
+    findinfiles.set_search_text("spam")
+    findinfiles.set_directory(osp.join(LOCATION, "data"))
+    findinfiles.set_conf('max_results', 6)
+
+    with qtbot.waitSignal(findinfiles.sig_max_results_reached):
+        findinfiles.find()
+
+    # Assert the results are the expected ones to reproduce the bug this test
+    # tries to catch. In other words, this is here to prevent future changes to
+    # our test files that would render this test useless.
+    results = {
+        'spam.py': [(2, 7), (5, 1), (7, 12)],
+        'spam.cpp': [(2, 9), (6, 15), (8, 2)]
+    }
```

Comment:
Seems like the results depend on the OS @ccordoba12 ? Seems like on Windows all the file occurences listed as results are from `spam.cpp` while on Linux and MacOS the results include occurences on `spam.py` and `spam.txt`. Maybe is there some other config for the plugin that is different depending on the OS to organize/sort the results or the iteration over the files or the way files are being iterated?

Reply:
I made the necessary the changes to make the test past on CIs and skip it on Windows (I don't know why it works differently there).

In addition, I verified that the test still fails in `5.x` by showing file items without matches.